x-pack/filebeat/input/azureeventhub/config.go

```diff
@@ -10,26 +10,72 @@ import (
 	"errors"
 	"fmt"
 	"strings"
+	"time"
 	"unicode"
 
 	"github.com/elastic/elastic-agent-libs/logp"
 )
 
+const ephContainerName = "filebeat"
+
 type azureInputConfig struct {
+	// EventHubName is the name of the event hub to connect to.
+	EventHubName string `config:"eventhub" validate:"required"`
+	// ConnectionString is the connection string to connect to the event hub.
 	ConnectionString string `config:"connection_string" validate:"required"`
-	EventHubName     string `config:"eventhub" validate:"required"`
-	ConsumerGroup    string `config:"consumer_group"`
+	// ConsumerGroup is the name of the consumer group to use.
+	ConsumerGroup string `config:"consumer_group"`
 	// Azure Storage container to store leases and checkpoints
-	SAName      string `config:"storage_account"`
-	SAKey       string `config:"storage_account_key"`
+	SAName string `config:"storage_account" validate:"required"`
+	// SAKey is used to connect to the storage account (processor v1 only)
+	SAKey string `config:"storage_account_key"`
+	// SAConnectionString is used to connect to the storage account (processor v2 only)
+	SAConnectionString string `config:"storage_account_connection_string"`
+	// SAContainer is the name of the storage account container to store
+	// partition ownership and checkpoint information.
 	SAContainer string `config:"storage_account_container"`
 	// by default the azure public environment is used, to override, users can provide a specific resource manager endpoint
 	OverrideEnvironment string `config:"resource_manager_endpoint"`
 	// cleanup the log JSON input for known issues, options: SINGLE_QUOTES, NEW_LINES
 	SanitizeOptions []string `config:"sanitize_options"`
+	// Controls if the input should perform the checkpoint information
+	// migration from v1 to v2 (processor v2 only).
+	MigrateCheckpoint bool `config:"migrate_checkpoint"`
+	// Processor version to use (v1 or v2). Default is v1.
+	ProcessorVersion string `config:"processor_version"`
+	//
+	ProcessorUpdateInterval time.Duration `config:"processor_update_interval"`
+	// Controls the start position for all partitions (processor v2 only).
+	ProcessorStartPosition string `config:"processor_start_position"`
+	// Processor receive timeout (processor v2 only).
+	// Wait up to `PartitionReceiveTimeout` for `PartitionReceiveCount` events,
+	// otherwise returns whatever we collected during that time.
+	PartitionReceiveTimeout time.Duration `config:"partition_receive_timeout"`
+	// Processor receive count (processor v2 only).
+	// Wait up to `PartitionReceiveTimeout` for `PartitionReceiveCount` events,
+	// otherwise returns whatever we collected during that time.
+	PartitionReceiveCount int `config:"partition_receive_count"`
```

Comment:
Add config's Validation in `Validate` method?

Reply:
The config already have a `Validate()` method. Are there specific checks we should add there?

Reply:
For the types you added that is either `time.Duration` / `int` to make sure the input is never negative `(<0)`. Example: ProcessorUpdateInterval, PartitionReceiveTimeout, and PartitionReceiveCount

Reply:
I added the missing checks to validate the boundaries for the new options' values. I also updated some option descriptions for consistency and default value.

Reply:
Thanks!